cfme/infrastructure/virtual_machines.py

```diff
@@ -1307,3 +1307,12 @@ class VmEngineRelationship(CFMENavigateStep):
     def step(self):
         self.prerequisite_view.toolbar.configuration.item_select(
             'Edit Management Engine Relationship')
+
+
+@navigator.register(Vm, 'Explore')
+class VmExplore(CFMENavigateStep):
```

Comment:
**REQUIRED**
I don't understand what's happening here. You are navigating to the list of all virtual machines, but `InfraVmDetailsView` will be returned. Please use one of these already implemented steps: [`VmAllWithTemplates`](https://github.com/ManageIQ/integration_tests/blob/master/cfme/infrastructure/virtual_machines.py#L1136) or [`VmAllWithTemplatesDetails`](https://github.com/ManageIQ/integration_tests/blob/master/cfme/infrastructure/virtual_machines.py#L1171)

Reply:
I have navigated to the all page because I want to collect all the VM names from the accordion and then navigate to one of the VmDetails page by clicking from the options available on the accordion to validate my test as pass or fail.
I also require InfraVmDetails view because it provides me the access to the accordion and also it gives the title of the page used in the below lines:
value = view.sidebar.vmstemplates.tree.read_contents()
return "Instance" in view.title.text

Reply:
For me it's a misusing of the navigation. Please use one of the aforementioned steps and when you perform `click_path` in the tree just create a required view with `create_view` method.

Reply:
Yes sure, I can certainly do it that way :)